### PR TITLE
[Documentation] database_files_update.pl perldocification 

### PR DESCRIPTION
**Pull request by cmadjar**
First, moved this script under the tools directory as it is an (old) tool scripts.

Second, using perldoc/perlpod to document `database_files_update.pl` so that users can type in the terminal
`perldoc database_files_update.pl` and get the documentation.

Using the `pod2markdown` tool from perl `Pod::Markdown` module, the `database_files_update.pl` file has been created so that we have a web displayed version of the documentation.
`pod2markdown database_files_update.pl > database_files_update.md`

Dependencies added: perldoc, Pod::Markdown

### CODE REVIEW

File: docs/scripts_md/database_files_update.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+perl database\_files\_update.pl `[options]`
+
+Available option is:
+
+\-profile: name of the config file in ../dicom-archive/.loris\_mri
+
+# DESCRIPTION
+
+This script updates the path stored in the `files` and `parameter_file`
+tables to remove the <\\$data\_dir> part of the path for security improvements.
```
Comment (MounaSafiHarab) [inline docs/scripts_md/database_files_update.md]: the `$data_dir` looks weird in the .md file.

File: docs/scripts_md/database_files_update.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# NAME
+
+database\_files\_update.pl -- Updates path stored in `files` and
+`parameter_file` tables so that they are relative to data\_dir
```
Comment (MounaSafiHarab) [inline docs/scripts_md/database_files_update.md]: maybe use C<> for the `$data_dir`?

File: docs/scripts_md/database_files_update.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+This script updates the path stored in the `files` and `parameter_file`
+tables to remove the <\\$data\_dir> part of the path for security improvements.
+
+## Methods
+
+### get\_minc\_files($data\_dir, $dbh)
+
+Gets the list of MINC files to update the location in the `files` table.
+
+INPUT: data directory from the `Config` tables, database handle
```
Comment (MounaSafiHarab) [inline docs/scripts_md/database_files_update.md]: INPUTS (same comment applies to all INPUTS below)

File: docs/scripts_md/database_files_update.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+Updates the location of MINC files in the `files` table.
+
+INPUT: File ID, new MINC relative location, database handle
+
+RETURNS: Number of rows affected by the update (should always be 1)
+
+### get\_parameter\_files($data\_dir, $parameter\_type, $dbh)
+
+Gets list of JIV files to update location in `parameter_file` (remove root
+directory from path)
```
Comment (MounaSafiHarab) [inline docs/scripts_md/database_files_update.md]: I would reword by saying ".... in the `parameter_file` table by removing the root directory from the path"

File: docs/scripts_md/database_files_update.md (hunk 2 above)
Comment (cmadjar) [reply docs/scripts_md/database_files_update.md]: good catch!

File: docs/scripts_md/database_files_update.md (hunk 4 above)
Comment (cmadjar) [reply docs/scripts_md/database_files_update.md]: done

File: docs/scripts_md/database_files_update.md (hunk 3 above)
Comment (cmadjar) [reply docs/scripts_md/database_files_update.md]: done

File: docs/scripts_md/database_files_update.md (hunk 1 above)
Comment (cmadjar) [reply docs/scripts_md/database_files_update.md]: should be fixed now.

File: tools/database_files_update.pl [hunk 5]
```diff
@@ -18,7 +18,7 @@ =head1 SYNOPSIS
 =head1 DESCRIPTION
 
 This script updates the path stored in the C<files> and C<parameter_file>
-tables to remove the <\$data_dir> part of the path for security improvements.
+tables to remove the C<$data_dir> part of the path for security improvements.
```
Comment (MounaSafiHarab) [inline tools/database_files_update.pl]: just a tiny tiny comment (data_dir on this line and 10 lines earlier is inconsistent; see screenshot)... I don't mind either way, but it might be good to make them look the same!
data_dir

**Comment by MounaSafiHarab** [comment]
@cmadjar 

have you tried this script? It does not do anything to me! I manually changed one file in the DB to specifically read the $data_dir in the beginning, and nothing happens!

**Comment by cmadjar** [comment]
Perfect. I am integrating your other comment for the data_dir and it will be ready to be merged :)

**Comment by cmadjar** [comment]
@MounaSafiHarab I integrated the changes you suggested. I will send another PR for the bugfix you found. Thanks!
